src/compiler/transformers/classFields.ts

```diff
@@ -632,10 +632,27 @@ namespace ts {
 
             resumeLexicalEnvironment();
 
-            let indexOfFirstStatement = 0;
+            const needsSyntheticConstructor = !constructor && isDerivedClass;
+            let indexOfFirstStatementAfterSuper = 0;
+            let indexAfterLastPrologueStatement = 0;
+            let foundSuperStatement = false;
             let statements: Statement[] = [];
 
-            if (!constructor && isDerivedClass) {
+            if (constructor) {
+                const initialDirectivesAndSuperCall = addPrologueDirectivesAndInitialSuperCall(factory, constructor, statements, visitor);
+                indexOfFirstStatementAfterSuper = initialDirectivesAndSuperCall.indexOfFirstStatementAfterSuper;
+                indexAfterLastPrologueStatement = initialDirectivesAndSuperCall.indexAfterLastPrologueStatement;
+                foundSuperStatement = initialDirectivesAndSuperCall.foundSuperStatement;
+
+                // Add existing statements before any super call
+                statements.splice(

```

Comment:
I'm not a huge fan of using `splice`. It'd be better if we can rework this such that calling `splice` isn't required.